pypeit/scripts/collate_1d.py

```diff
@@ -286,6 +287,69 @@ def build_coadd_file_name(source_object):
     
     return f'{coord_portion}_{instrument_name}_{date_portion}.fits'
 
+def refframe_correction(par, spectrograph, spec1d_files, spec1d_failure_msgs):
+
+    refframe = par['collate1d']['refframe']
+    msgs.info(f"Performing a {refframe} correction")
+
+    for spec1d in spec1d_files:
+        # Get values from the fits header needed to calculate the correction
+        try:
+            sobjs = SpecObjs.from_fitsfile(spec1d)
+            hdr_ra = sobjs.header['RA']
+            hdr_dec = sobjs.header['DEC']
+            hdr_radec = ltu.radec_to_coord((hdr_ra, hdr_dec))
+            obstime = Time(sobjs.header['MJD'], format='mjd')
+        except Exception as e:
+            msg = f'Failed to perform {refframe} correction on {spec1d}: {e}'
+            msgs.info(msg)
+            spec1d_failure_msgs.append(msg)
+            continue
+
+        corrected_at_least_one = False
+        for sobj in sobjs:
+            if sobj['VEL_CORR'] is not None:
+                # Don't double correct
+                msg = f"Not performing {refframe} correction for {spec1d} object {sobj['NAME']} because it has already been corrected."
+                msgs.info(msg)
+                spec1d_failure_msgs.append(msg)
+                continue
+            # Use the SpecObj RA/DEC if it's available, otherwise use the value from the header
+            if sobj['RA'] is not None and sobj['DEC'] is not None:
+                radec = ltu.radec_to_coord((sobj['RA'], sobj['DEC']))
+            else:
+                radec = hdr_radec
+            # Calculate the correction
+            vel, vel_corr = wave.geomotion_correct(radec, obstime,
+                                                   spectrograph.telescope['longitude'],
+                                                   spectrograph.telescope['latitude'],
+                                                   spectrograph.telescope['elevation'],
+                                                   refframe)
+            # Apply correction to objects
+            msgs.info(f'Applying {refframe} correction = {vel} km/s')
+            sobj.apply_helio(vel_corr, refframe)
+            corrected_at_least_one = True
+        if corrected_at_least_one:
+            sobjs.write_to_fits(subheader = sobjs.header, outfile=spec1d, overwrite=True)
+
+def copy_spec1d_to_outdir(spec1d_files, outdir):
```

Comment:
Could you treat the `spec1d_outdir` parameter like the `outdir` parameter, i.e., if the directory dos not exists, create it instead of crashing?

Reply:
Yes, that was an oversight on my part. Fixed.